File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
void printi(int i);

int main(void)
{
    int no_of_words = 1, no_of_letters = 0, no_of_sent = 1, I, p = 1;
    float L, S;
    string text = get_string("Text: ");
    int length = strlen(text);
    for (int i = 0; i < length; i++)
    {

        if (text[i] == ' ' || text[i] == '\n')
        {
            no_of_words += 1;
        }
        else if (text[i] == '.' || text[i] == '?')
        {
            no_of_sent += 1;
        }

        else if (isalpha(text[i]) != 0)
        {
            no_of_letters += 1;
        }
    }

    L = 100.00 * (float) no_of_letters / (float) no_of_words;
    S = 100.00 * (float) no_of_sent / (float) no_of_words;
    I = (round) ((0.0588 * L) - (0.296 * S) - 15.8);
    if (I < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (I >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", I + 1);
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
word_count = 1
sentence_count = 0
letter_count = 0
for l in text:
    if l == ' ':
        word_count += 1

    elif l == '.' or l == '!' or l == '?':
        sentence_count += 1

    elif l.isalpha():
        letter_count += 1

L = (letter_count/word_count) * 100
S = (sentence_count/word_count) * 100

index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")

elif index > 16:
    print("Grade 16+")

else:
    print(f"Grade {index}")
